### DISCUSSION TITLE
how to encrypt a field of a model in laravel ?

### DISCUSSION THREAD

**Discussion by lakoud** [Schweinepriester/achievement-testarea#54] [Q&A]
how to encrypt a field of a model in laravel ?

**Comment by rubikproxy** [comment]
1. Add the **`Crypt` facade** in the Laravel model to use Laravel's built-in encryption.  
2. Define an **accessor** in the model to decrypt the field when retrieving it from the database.  
3. Define a **mutator** in the model to encrypt the field before saving it to the database.  
4. Use **`Crypt::encryptString()`** in the mutator to encrypt the value.  
5. Use **`Crypt::decryptString()`** in the accessor to decrypt the value.  
6. Ensure the **APP_KEY** in the `.env` file is securely configured because Laravel encryption relies on it.

**Comment by giangeralcus** [comment]
To encrypt a model field in Laravel, use encrypted casting (built-in, Laravel 7+):
1. In your model, add the cast:
protected $casts = ['secret_field' => 'encrypted'];
2. Done - Laravel now encrypts on save and decrypts on read automatically using your APP_KEY.
If you need to search on the field, note regular encrypted casts are not searchable (different ciphertext each save) - use a blind-index column for lookups. For existing data, run a one-time migration that loops over rows and re-saves them so they get encrypted.
